functions/src/triggers/OrderUpdate.ts

```diff
@@ -0,0 +1,100 @@
+import { Order } from "../types";
+
+import * as functions from "firebase-functions";
+import { orderCollection } from "../database";
+import { getApis } from "../apis";
+import { Address, CreateResponse } from "../lob";
+import { notifyUserOnLetterCreate } from "../emails";
+import { makeLetter } from "../letters";
+
+export const markOrderPaid = async (orderId: string) => {
+  console.log("marking order paid", orderId);
+  const docs = await orderCollection.where("orderId", "==", orderId).get();
+  if (docs.empty) {
+    throw new Error("no order with id " + orderId);
+  }
+
+  const order = docs.docs[0];
+
+  const orderPromise = order.ref.update({ paid: true });
+  return orderPromise;
+};
+
+export const executeOrderId = async (orderId: string) => {
+  console.log("executing order", orderId);
+
+  const docs = await orderCollection.where("orderId", "==", orderId).get();
+  if (docs.empty) {
+    throw new Error("no order with id " + orderId);
+  }
+
+  const order = docs.docs[0];
+  const orderData: Order = order.data() as Order;
+  return executeOrder(orderData);
+};
+
+export const executeOrder = async (orderData: Order): Promise<any> => {
+  if (!orderData || !orderData.toAddresses || !orderData.body) {
+    throw new Error("no order with id ");
+  }
+
+  const apis = getApis({isTest: orderData.test});
+
+  const lobPromises = orderData.toAddresses.map((toAddress: Address) => {
+    return new Promise((resolve, reject) => {
+      apis.lob.letters.create(
+        {
+          description: "Demo Letter",
```

Comment:
Can you explain why this is all test/demo?

Reply:
re: demo - that was just laziness from copypasta, changed the description

test is commented pretty heavily elsewhere but I'll add a little more - if the API request is made with test=true it uses test envs for stripe and lob